File: pbworkspacewcl/src/app/dashboard/services/dashboard.service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ConfigService } from 'src/app/core/services/config.service';
import { WidgetDTO } from '../commons/dto/widget-dto';
import { BandoWidgetDTO } from '../commons/dto/bando-widget-dto';
import { EsitoOperazioneDTO } from 'src/app/shared/commons/dto/esito-operazione-dto';

@Injectable({
  providedIn: 'root'
})
export class DashboardService {

  constructor(
    private configService: ConfigService,
    private http: HttpClient
  ) { }

  isDashboardVisible(): Observable<boolean> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/isDashboardVisible';
    return this.http.get<boolean>(url);
  }

  areWidgetsConfigured(): Observable<boolean> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/areWidgetsConfigured';
    return this.http.get<boolean>(url);
  }

  getWidgets(): Observable<Array<WidgetDTO>> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/widgets';
    return this.http.get<Array<WidgetDTO>>(url);
  }

  getBandiDaAssociare(): Observable<Array<BandoWidgetDTO>> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/bandiDaAssociare';
    return this.http.get<Array<BandoWidgetDTO>>(url);
  }

  getBandiAssociati(idWidget: number): Observable<Array<BandoWidgetDTO>> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/bandiAssociati';
    let params = new HttpParams().set("idWidget", idWidget.toString());
    return this.http.get<Array<BandoWidgetDTO>>(url, { params: params });
  }

  changeWidgetAttivo(idWidget: number, flagWidgetAttivo: boolean): Observable<EsitoOperazioneDTO> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/widgetAttivo';
    let params = new HttpParams().set("idWidget", idWidget.toString()).set("flagWidgetAttivo", flagWidgetAttivo.toString());
    return this.http.post<EsitoOperazioneDTO>(url, { body: null }, { params: params });
  }

  associaBandoAWidget(idWidget: number, progrBandoLineaIntervento: number): Observable<EsitoOperazioneDTO> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/associaBandoAWidget';
    let params = new HttpParams().set("idWidget", idWidget.toString()).set("progrBandoLineaIntervento", progrBandoLineaIntervento.toString());
    return this.http.post<EsitoOperazioneDTO>(url, { body: null }, { params: params });
  }

  disassociaBandoAWidget(idBandoLinSoggWidget: number): Observable<EsitoOperazioneDTO> {
    let url = this.configService.getApiURL() + '/restfacade/dashboard/disassociaBandoAWidget';
    let params = new HttpParams().set("idBandoLinSoggWidget", idBandoLinSoggWidget.toString());
    return this.http.delete<EsitoOperazioneDTO>(url, { params: params });
  }
}

File: pbworkspace/pbworkspace-jar/src/main/java/it/csi/pbandi/pbworkspace/business/service/DashboardService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbworkspace.business.service;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.DELETE;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/dashboard")
public interface DashboardService {

	@GET
	@Path("isDashboardVisible")
	@Produces(MediaType.APPLICATION_JSON)
	public Response isDashboardVisible(@Context HttpServletRequest req) throws Exception;

	@GET
	@Path("areWidgetsConfigured")
	@Produces(MediaType.APPLICATION_JSON)
	public Response areWidgetsConfigured(@Context HttpServletRequest req) throws Exception;

	@GET
	@Path("widgets")
	@Produces(MediaType.APPLICATION_JSON)
	public Response getWidgets(@Context HttpServletRequest req) throws Exception;

	@GET
	@Path("bandiDaAssociare")
	@Produces(MediaType.APPLICATION_JSON)
	public Response getBandiDaAssociare(@Context HttpServletRequest req) throws Exception;

	@GET
	@Path("bandiAssociati")
	@Produces(MediaType.APPLICATION_JSON)
	public Response getBandiAssociati(@QueryParam("idWidget") Long idWidget, @Context HttpServletRequest req)
			throws Exception;

	@POST
	@Path("widgetAttivo")
	@Produces(MediaType.APPLICATION_JSON)
	public Response changeWidgetAttivo(@QueryParam("idWidget") Long idWidget,
			@QueryParam("flagWidgetAttivo") Boolean flagWidgetAttivo, @Context HttpServletRequest req) throws Exception;

	@POST
	@Path("associaBandoAWidget")
	@Produces(MediaType.APPLICATION_JSON)
	public Response associaBandoAWidget(@QueryParam("idWidget") Long idWidget,
			@QueryParam("progrBandoLineaIntervento") Long progrBandoLineaIntervento, @Context HttpServletRequest req)
			throws Exception;

	@DELETE
	@Path("disassociaBandoAWidget")
	@Produces(MediaType.APPLICATION_JSON)
	public Response disassociaBandoAWidget(@QueryParam("idBandoLinSoggWidget") Long idBandoLinSoggWidget,
			@Context HttpServletRequest req) throws Exception;

}
